File: ios/Classes/utils/UnifiedAssets.m
//
// Created by Yohom Bao on 2018-12-01.
//

#import "UnifiedAssets.h"
#import "AMapBasePlugin.h"

static const NSString *PACKAGE = @"packages/amap_base/";

@implementation UnifiedAssets {

}
+ (NSString *)getAssetPath:(NSString *)asset {
    NSString *key = [AMapBasePlugin.registrar lookupKeyForAsset:[NSString stringWithFormat:@"%@%@", PACKAGE, asset]];
    return [[NSBundle mainBundle] pathForResource:key ofType:nil];
}

@end

File: android/src/main/kotlin/me/yohom/amapbase/utils/UnifiedAssets.kt
package me.yohom.amapbase.utils

import android.graphics.BitmapFactory
import com.amap.api.maps.model.BitmapDescriptor
import com.amap.api.maps.model.BitmapDescriptorFactory
import me.yohom.amapbase.AMapBasePlugin

const val PACKAGE = "packages/amap_base/"

object UnifiedAssets {
    private val assetManager = AMapBasePlugin.registrar.context().assets

    fun getBitmapDescriptor(asset: String): BitmapDescriptor {
        val assetFileDescriptor = assetManager.openFd(AMapBasePlugin.registrar.lookupKeyForAsset(asset, PACKAGE))
        return BitmapDescriptorFactory.fromBitmap(BitmapFactory.decodeStream(assetFileDescriptor.createInputStream()))
    }
}